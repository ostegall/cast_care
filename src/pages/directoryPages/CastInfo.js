import { Text, View, Pressable } from 'react-native';
import styles from '../../styles';

export default function CastInfo({navigation}) { //Page with general cast upkeep
    return (
      <View style={{flex: 1, justifyContent: 'center'}}>
        <Text>
          {'\u2022'} {/* code for a bullet point */}
            <Text style={styles.page_text}>
              Keep cast clean and dry
            </Text>
        </Text>
        <Text>
          {'\u2022'}
            <Text style={styles.page_text}>
              Do not stick anything inside of cast or splint
            </Text>
        </Text>
        <Text>
          {'\u2022'}
            <Text style={styles.page_text}>
              If cast/splint becomes itchy: blow dry on low cool air, tap, or itch other side
            </Text>
        </Text>
        <Text>
          {'\u2022'}
            <Text style={styles.page_text}>
              Elevate above the level of patient's heart
            </Text>
        </Text>
        <Text>
          {'\u2022'}
            <Text style={styles.page_text}>
              It is normal to have increased pain right after a cast/splint is put on due to moving the fracture
            </Text>
        </Text>
        <Text>
          {'\u2022'}
            <Text style={styles.page_text}>
              Patient can ice through the cast/splint by using a gel ice pack or making sure the condensation from the ice bag does not get on the cast/splint; some people may or may not be able to feel the ice
            </Text>
        </Text>
        <Text>
          {'\u2022'}
            <Text style={styles.page_text}>
              No powders or lotions down cast/splint
            </Text>
        </Text>
        <Text>
          {'\u2022'}
            <Text style={styles.page_text}>
              Wiggle fingers as much as possible
            </Text>
        </Text>
      </View>
    )
  }