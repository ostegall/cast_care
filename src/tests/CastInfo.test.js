import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'

import CastInfo from '../pages/directoryPages/CastInfo';

//Makes sure app component renders correctly without crashing
it("renders without crashing", () => {
  shallow(<CastInfo />);
});

//Snapshot test to ensure UI stays consistent
it('renders correctly', () => {
  const tree = renderer.create(<CastInfo />).toJSON();
  expect(tree).toMatchSnapshot();
});