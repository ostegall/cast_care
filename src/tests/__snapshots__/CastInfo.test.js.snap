// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    •
     
    <Text
      style={
        Object {
          "alignItems": "center",
          "fontSize": 17,
          "letterSpacing": 0.5,
          "marginBottom": 5,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 5,
        }
      }
    >
      Keep cast clean and dry
    </Text>
  </Text>
  <Text>
    •
    <Text
      style={
        Object {
          "alignItems": "center",
          "fontSize": 17,
          "letterSpacing": 0.5,
          "marginBottom": 5,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 5,
        }
      }
    >
      Do not stick anything inside of cast or splint
    </Text>
  </Text>
  <Text>
    •
    <Text
      style={
        Object {
          "alignItems": "center",
          "fontSize": 17,
          "letterSpacing": 0.5,
          "marginBottom": 5,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 5,
        }
      }
    >
      If cast/splint becomes itchy: blow dry on low cool air, tap, or itch other side
    </Text>
  </Text>
  <Text>
    •
    <Text
      style={
        Object {
          "alignItems": "center",
          "fontSize": 17,
          "letterSpacing": 0.5,
          "marginBottom": 5,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 5,
        }
      }
    >
      Elevate above the level of patient's heart
    </Text>
  </Text>
  <Text>
    •
    <Text
      style={
        Object {
          "alignItems": "center",
          "fontSize": 17,
          "letterSpacing": 0.5,
          "marginBottom": 5,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 5,
        }
      }
    >
      It is normal to have increased pain right after a cast/splint is put on due to moving the fracture
    </Text>
  </Text>
  <Text>
    •
    <Text
      style={
        Object {
          "alignItems": "center",
          "fontSize": 17,
          "letterSpacing": 0.5,
          "marginBottom": 5,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 5,
        }
      }
    >
      Patient can ice through the cast/splint by using a gel ice pack or making sure the condensation from the ice bag does not get on the cast/splint; some people may or may not be able to feel the ice
    </Text>
  </Text>
  <Text>
    •
    <Text
      style={
        Object {
          "alignItems": "center",
          "fontSize": 17,
          "letterSpacing": 0.5,
          "marginBottom": 5,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 5,
        }
      }
    >
      No powders or lotions down cast/splint
    </Text>
  </Text>
  <Text>
    •
    <Text
      style={
        Object {
          "alignItems": "center",
          "fontSize": 17,
          "letterSpacing": 0.5,
          "marginBottom": 5,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 5,
        }
      }
    >
      Wiggle fingers as much as possible
    </Text>
  </Text>
</View>
`;
